internal/connect/command_offset_update.go

```diff
@@ -59,58 +60,65 @@ func (c *offsetCommand) alterOffset(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	connectorIdToName, err := c.mapConnectorIdToName(environmentId, kafkaCluster.ID)
+	connector, err := c.V2Client.GetConnectorExpansionById(args[0], environmentId, kafkaCluster.ID)
 	if err != nil {
 		return err
 	}
 
+	connectorName := connector.Info.GetName()
+
 	request, err := c.getAlterOffsetRequestBody(cmd)
 	if err != nil {
 		return err
 	}
 
-	_, err = c.V2Client.AlterConnectorOffsets(connectorIdToName[args[0]], environmentId, kafkaCluster.ID, *request)
+	_, err = c.V2Client.AlterConnectorOffsets(connectorName, environmentId, kafkaCluster.ID, *request)
 	if err != nil {
 		return err
 	}
 
 	var offsetStatus connectv1.ConnectV1AlterOffsetStatus
-	currTime := time2.Now()
 	table := output.NewTable(cmd)
-	for {
-		offsetStatus, err = c.V2Client.AlterConnectorOffsetsRequestStatus(connectorIdToName[args[0]], environmentId, kafkaCluster.ID)
+
+	err = retry.Retry(time.Second, 30*time.Second, func() error {
+		offsetStatus, err = c.V2Client.AlterConnectorOffsetsRequestStatus(connectorName, environmentId, kafkaCluster.ID)
 		if err != nil {
 			return err
 		}
 
-		if (offsetStatus.GetStatus().Phase != "PENDING" || time2.Since(currTime).Seconds() > 30) && &offsetStatus != nil {
-			var appliedAt string
-			if offsetStatus.AppliedAt.IsSet() {
-				appliedAt = offsetStatus.AppliedAt.Get().String()
-			}
-
-			var phase string
-			var message string
-			_, isStatusSet := offsetStatus.GetStatusOk()
-			if isStatusSet {
-				phase = offsetStatus.GetStatus().Phase
-				if messagePtr := offsetStatus.GetStatus().Message; messagePtr != nil {
-					message = *messagePtr
-				}
-			}
-			table.Add(&alterStatusOut{
-				Id:        args[0],
-				Phase:     phase,
-				Message:   message,
-				AppliedAt: appliedAt,
-			})
-			return table.Print()
+		if offsetStatus.GetStatus().Phase != "PENDING" {
+			return nil
+		}
+		return nil
```

Comment:
It looks like we always return `nil`. Why do we have this `if` statement, then?

Reply:
Updated